common/src/main/java/io/crate/common/collections/Maps.java

```diff
@@ -33,6 +34,19 @@
 
 public final class Maps {
 
+    public static <K, V> Map<K, V> concat(Map<K, V> m1, Map<K, V> m2) {
+        if (m1.isEmpty()) {
+            return m2;
+        }
+        if (m2.isEmpty()) {
+            return m1;
+        }
+        HashMap<K, V> result = new HashMap<>();
```

Comment:
this loses the predictable iteration order of the sources and scrambles the result. I think `concat` should either work with hash maps only or be renamed to `concatAndHash` or something like that